eduHub/src/main/kotlin/com/arsiu/eduhub/config/bpp/NotifyTriggerAnnotationHandlerBeanPostProcessor.kt

```diff
@@ -62,14 +62,14 @@ class NotifyTriggerAnnotationHandlerBeanPostProcessor(
         return beanClass.memberFunctions.firstOrNull {
             (it.name == method.name)
                     &&
-                    (it.javaClass.typeParameters.contentEquals(method.javaClass.typeParameters))
+            (it.javaClass.typeParameters.contentEquals(method.javaClass.typeParameters))
         }?.findAnnotation<NotifyTrigger>()
     }
 
     @OptIn(DelicateCoroutinesApi::class)
     private fun notifyUsers(msg: String) {
         GlobalScope.launch {
-            userService.findAll()
+            userService.findAll().collectList().block()!!
```

Comment:
Suggestion:
```kotlin
userService.findAll()
    .parallel()
    .runOn(<Some-scheduler-here>)
    .filter { it.role == Role.STUDENT }
    .doOnNext { 
        emailServiceImpl.sendMail(EmailParameters(it.email, msg, msg)) 
    }
    .subscribe()
```  

So, each email could be sent in a separate thread. This code doesn't require usage of coroutinies.

Reply:
`+`